### PR TITLE
Add metadata generation job

### PR DESCRIPTION
**Pull request by artonge**

### CODE REVIEW

**Review by ArtificialOwl — APPROVED** [review]

File: core/BackgroundJobs/GenerateMetadataJob.php [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+		if ($folder->getMountPoint() instanceof \OC\Files\Mount\MoveableMount) {
+			return;
+		}
+
+		foreach ($folder->getDirectoryListing() as $node) {
+			if ($node instanceof Folder) {
+				$this->scanFolder($node);
+				continue;
+			}
+
```
Comment (ArtificialOwl) [inline core/BackgroundJobs/GenerateMetadataJob.php:104]: what if you `getMetadata()` first and see if it exists or not, and `continue;` if it exists ?

Comment (artonge) [reply core/BackgroundJobs/GenerateMetadataJob.php:104]: It could be partial ?

Comment (ArtificialOwl) [reply core/BackgroundJobs/GenerateMetadataJob.php:104]: partially generated ? it's fine, the script is not about updating metadata but creating

Comment (artonge) [reply core/BackgroundJobs/GenerateMetadataJob.php:104]: The script is about filling the metadata with useful content. If the metadata are partial, then it is not as useful.

**Review by come-nc — APPROVED** [review]

**Pull request merged**